### DISCUSSION TITLE
Where is the source for vec_lite.js?

### DISCUSSION THREAD

**Discussion by delano** [brianpetro/obsidian-smart-connections#267] [General]
Hey, great work on this plugin. I use it daily to navigate my vaults. I'm looking into contributing and have a couple questions. 

### Context

There are references in the repo to `vec_lite/vec_lite.js` and `VecLite` / `VecLite2` classes:

**package.json**
https://github.com/brianpetro/obsidian-smart-connections/blob/d34f90d17c7df692e60fa3452ef4ef7857bc9cd1/package.json#L31-L33

**dist/main.js**
https://github.com/brianpetro/obsidian-smart-connections/blob/d34f90d17c7df692e60fa3452ef4ef7857bc9cd1/dist/main.js#L6-L11

**src/index.js**
https://github.com/brianpetro/obsidian-smart-connections/blob/d34f90d17c7df692e60fa3452ef4ef7857bc9cd1/src/index.js#L1-L2

https://github.com/brianpetro/obsidian-smart-connections/blob/d34f90d17c7df692e60fa3452ef4ef7857bc9cd1/src/index.js#L197-L199

The repo doesn't appear to include the original source files that are being referenced here though (in either javascript or typescript). 

### Contributing 

This presents a challenge when it comes to submitting code contributions. We're only able to work with the js files generated during a build but not directly with the source. This must present a challenge when accepting PRs too where presumably you'd need to copy the changes over manually, re-build, and then push the changes to the generated files to make them available in the repo.

If it's intentional (which is totally fine), you may want to look into using an alternative license since [GPLv3](https://github.com/brianpetro/obsidian-smart-connections/blob/d34f90d17c7df692e60fa3452ef4ef7857bc9cd1/LICENSE#L1-L6) requires include the original source. If they just haven't been committed yet (which is also totally fine), please do so myself and others can contribute work to the code itself 🙏

**Comment by brianpetro** [comment]
@delano Thanks for your interest in Smart Connections and contributing to the project.

In addition to "VecLite," Smart Connections is being refactored into modules that will consist of the "AI" logic, leaving this Smart Connections plugin to represent the Obsidian-specific pieces.

At this point, I do intend to release all of the modules. However, the intent behind Smart Connections was to contribute to the Obsidian community, and not necessarily the open-source community (though that would be great, too).

Regarding the license, it's purposefully restrictive to give me options as I decide how to proceed.

In the meantime, the best way to contribute would be detailed feature requests, including code snippets if you're so inclined.

I appreciate your feedback and will be happy to consider any additional insights you might be able to provide.

Thanks for helping make Smart Connections the best it can be,
Brian 🌴
